### PR TITLE
[1LP][RFR] Provider object should hold endpoints in dict instead of list

### PR DESCRIPTION
**Pull request by izapolsk**
every attempt to find right endpoint causes some overhead when endpoints are stored in list. 
So, according to @psav advice, I moved endpoints to dict where dict's structure looks like
endoints = {endpoint.name: endpoint}

{{pytest: cfme/tests/infrastructure/test_esx_direct_host.py cfme/tests/cloud/test_providers.py cfme/tests/infrastructure/test_providers.py }}

### CODE REVIEW

**Comment by izapolsk** [comment]
dict changes have been tested in PRT and results are successful. There was one test failed due to absent default endpoint but it has just been fixed and tested locally.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 2663be9e5a7a2c2042e2b8b312be708c57ca15b2:

`cfme/common/provider.py`:
- :bangbang: Line 175:101: **E501** *line too long (101 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/common/provider.py [hunk 1]
```diff
@@ -244,9 +249,9 @@ def update(self, updates, endpoints=None, cancel=False, validate_credentials=Tru
 
             # filling endpoints
             if endpoints:
-                endpoints = endpoints if isinstance(endpoints, list) else [endpoints, ]
+                endpoints = self._prepare_endpoints(endpoints)
 
-                for endpoint in endpoints:
+                for _, endpoint in endpoints.items():
```
Comment (mkoura) [inline cfme/common/provider.py]: OPTIONAL: why not ``for endpoint in endpoints.values():``?

Comment (izapolsk) [inline cfme/common/provider.py]: good point. because when I was writing for statement, I didn't exactly know whether I'd use key :)

**Pull request merged**